src/main/java/net/sf/jabref/gui/preftabs/PreferencesDialog.java

```diff
@@ -219,15 +208,38 @@ public PreferencesDialog(JabRefFrame parent) {
 
     private void updateAfterPreferenceChanges() {
         setValues();
-        Map<String, ExportFormat> customFormats = Globals.prefs.customExports.getCustomExportFormats(Globals.prefs,
+        Map<String, ExportFormat> customFormats = prefs.customExports.getCustomExportFormats(prefs,
                 Globals.journalAbbreviationLoader);
-        LayoutFormatterPreferences layoutPreferences = Globals.prefs
+        LayoutFormatterPreferences layoutPreferences = prefs
                 .getLayoutFormatterPreferences(Globals.journalAbbreviationLoader);
-        SavePreferences savePreferences = SavePreferences.loadForExportFromPreferences(Globals.prefs);
+        SavePreferences savePreferences = SavePreferences.loadForExportFromPreferences(prefs);
         ExportFormats.initAllExports(customFormats, layoutPreferences, savePreferences);
 
         frame.removeCachedEntryEditors();
-        Globals.prefs.updateEntryEditorTabList();
+        prefs.updateEntryEditorTabList();
+    }
+
+    private void storeAllSettings(){
+        // First check that all tabs are ready to close:
+        int count = main.getComponentCount();
+        Component[] comps = main.getComponents();
+        for (int i = 0; i < count; i++) {
```

Comment:
If you don't need the counter variable please use for each. `for (Component currentComponent : comps)`.